File: offer-cache-service/src/main/scala/pl/allezon/offercache/service/OfferEndpoint.scala
package pl.allezon.offercache.service

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.{GetMapping, PathVariable, RequestMapping, RestController}
import pl.allezon.offercache.domain.Offer
import pl.allezon.offercache.repository.OfferRepository

@RestController
@RequestMapping(Array("/offers"))
class OfferEndpoint @Autowired()(val offerRepository: OfferRepository) {

  @GetMapping(Array("/{id}"))
  def getOffer(@PathVariable id: String): Offer = offerRepository.findById(id)
}

File: offer-storage-service/src/main/java/pl/allezon/offerstorage/web/OfferEndpoint.java
package pl.allezon.offerstorage.web;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;
import pl.allezon.offerstorage.domain.Offer;
import pl.allezon.offerstorage.service.OfferService;

@RestController
public class OfferEndpoint {

    private final OfferService offerService;

    @Autowired
    public OfferEndpoint(OfferService offerService) {
        this.offerService = offerService;
    }

    @PutMapping("/offer")
    public void createOffer(@RequestBody Offer offer) {
        this.offerService.createOffer(offer);
    }
}
